library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2424,6 +2425,19 @@ impl<T, A: Allocator> VecDeque<T, A> {
     /// deque.insert(idx, num);
     /// assert_eq!(deque, &[0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 42, 55]);
     /// ```
+    ///
+    /// Performing binary search on an unsorted `VecDeque` generates
+    /// unspecified and meaningless results:
+    ///
+    /// ```
+    /// use std::collections::VecDeque;
+    ///
+    /// let deque: VecDeque<_> = [0, 13, 1, 1, 21, 3, 34].into();
+    ///
+    /// // 13 and 21 can't be found with binary search, even though they are in the VecDeque
```

Comment:
I think having doctests/examples to demonstrate unspecified behaviour might be a bit misleading, since it tests a specific behaviour (even if that behaviour is 'doesn't return desired results')